src/logics/Logic.cc

```diff
@@ -371,30 +383,28 @@ SRef Logic::declareUninterpretedSort(const std::string & name) {
     return getSort(sortSymbol, {});
 }
 
-PTRef Logic::resolveTerm(const char* s, vec<PTRef>&& args, char** msg) {
-    SymRef sref = term_store.lookupSymbol(s, args);
+PTRef Logic::resolveTerm(const char* s, vec<PTRef>&& args, SRef sortRef) {
+    SymRef sref = term_store.lookupSymbol(s, args, sortRef);
     if (sref == SymRef_Undef) {
         if (defined_functions.has(s)) {
             auto const & tpl = defined_functions[s];
-            try {
-                return instantiateFunctionTemplate(tpl, args);
-            } catch (OsmtApiException const & e) {
-                *msg = strdup(e.what());
-                return PTRef_Undef;
-            }
+            return instantiateFunctionTemplate(tpl, args);
         }
         else {
-            int written = asprintf(msg, "Unknown symbol `%s'", s);
-            assert(written >= 0); (void)written;
-            return PTRef_Undef;
+            std::string argSortsString;
+            for (int i = 0; i < args.size(); i++) {
+                PTRef tr = args[i];
+                argSortsString += printSort(getSortRef(tr)) + (i == args.size() - 1 ? "" : " ");
+            }
+            throw OsmtApiException("Unknown symbol `" + std::string(s) + ' ' + argSortsString + (sortRef != SRef_Undef ?  "/ " +printSort(sortRef) : "") + "'");
```

Comment:
```suggestion
            throw OsmtApiException("Unknown symbol `" + std::string(s) + ' ' + argSortsString + (sortRef != SRef_Undef ?  "/ " + printSort(sortRef) : "") + "'");
```
Just a missing space.